subsys/fs/littlefs_fs.c

```diff
@@ -600,8 +619,49 @@ static int littlefs_mount(struct fs_mount_t *mountp)
 	dev = flash_area_get_device(fs->area);
 	if (dev == NULL) {
 		LOG_ERR("can't get flash device: %s", log_strdup(fs->area->fa_dev_name));
-		ret = -ENODEV;
-		goto out;
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int littlefs_mount(struct fs_mount_t *mountp)
+{
+	int ret = 0;
+	struct fs_littlefs *fs = mountp->fs_data;
+	bool block_dev = littlefs_on_blkdev(mountp);
+
+	LOG_INF("LittleFS version %u.%u, disk version %u.%u",
+		LFS_VERSION_MAJOR, LFS_VERSION_MINOR,
+		LFS_DISK_VERSION_MAJOR, LFS_DISK_VERSION_MINOR);
+
+
+	if (block_dev) {
+#if IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV)
+		if (fs->blk_mounted) {
+			return -EBUSY;
+		}
+		if (disk_access_init(mountp->storage_dev) != 0) {
```

Comment:
Oh, I missed that.

Reply:
@jfischer-no 
Regarding your fist comment - having separate commit on Kconfig clarifies the code, as you have several small commits, which are then easy to port/debug.
Having the `FS_LITTLEFS_BLK_DEV` introduced when the feature is following is IMHO OK (at least is acceptable in other FOSS projects).

Regarding your second comment:
> There is difference how you handle the backends, probably because there is no clean separation between the backends and the common code.

As said before - this PR is to augment the current usage of the `littlefs` - i.e. to add disk device support. 
Before this PR not any single board was using `littlefs` with the SD card connected. My use case is clearly stated,
so I can test if this use case works. I don't have any interest in testing features which are out of my use case scope,
sorry.

> Here you initialize blk device and set fs->blkmounted, but do not lock file system mutex (fs_lock(fs);). For the raw flash backend, it just checked for busy here, but is initialized later after fs_lock(fs). Neither seems to be handled correct an are susceptible to race condition

I can move the `disk_access_init()` after the `fs_lock()`.

Reply:
@lmajewski, you should worry about things that are out of your scope.

IMO your PR would benefit from reusing the fs->area pointer in your changes as this would remove difference in handling between flash and sd backend. I would propose changing `const flash_area *area` into a `const void *backend` and to use this after casting to the appropiate format. For the flash this would point to the area, for the sd this would point to the disk.

Reply:
> Regarding your fist comment - having separate commit on Kconfig clarifies the code, as you have several small commits, which are then easy to port/debug.
> Having the `FS_LITTLEFS_BLK_DEV` introduced when the feature is following is IMHO OK (at least is acceptable in other FOSS projects).

You are replying to wrong comment, it does not belong here.

> > There is difference how you handle the backends, probably because there is no clean separation between the backends and the common code.
> 
> As said before - this PR is to augment the current usage of the `littlefs` - i.e. to add disk device support. Before this PR not any single board was using `littlefs` with the SD card connected. My use case is clearly stated, so I can test if this use case works. I don't have any interest in testing features which are out of my use case scope, sorry.

I do not get it.

> > Here you initialize blk device and set fs->blkmounted, but do not lock file system mutex (fs_lock(fs);). For the raw flash backend, it just checked for busy here, but is initialized later after fs_lock(fs). Neither seems to be handled correct an are susceptible to race condition
> 
> I can move the `disk_access_init()` after the `fs_lock()`.

Well, I think it is not enough. All the changes in `littlefs_mount()` should be revised, please look at my comments again.

Reply:
Or maybe even better, you could remove the fs->area pointer and use the `lcp->context` to validate if the backend is already mounted.